tests/unit/test_charm.py

```diff
@@ -0,0 +1,294 @@
+        self,
+        patch_push,
+        patch_exists,
+    ):
+        patch_exists.return_value = True
+
+        self.harness.container_pebble_ready(container_name="bessd")
+
+        patch_push.assert_not_called()
+
+    @patch("ops.model.Container.exec", new=Mock())
+    @patch("ops.model.Container.exists")
+    def test_given_bessd_config_file_is_written_when_bessd_pebble_ready_then_pebble_plan_is_applied(  # noqa: E501
+        self,
+        patch_exists,
+    ):
+        patch_exists.return_value = True
+
+        self.harness.container_pebble_ready(container_name="bessd")
+
+        expected_plan = {
+            "services": {
+                "bessd": {
+                    "startup": "enabled",
+                    "override": "replace",
+                    "command": "bessd -f -grpc-url=0.0.0.0:10514 -m 0",
+                    "environment": {"CONF_FILE": "/etc/bess/conf/upf.json"},
+                }
+            }
+        }
+
+        updated_plan = self.harness.get_container_pebble_plan("bessd").to_dict()
+
+        self.assertEqual(expected_plan, updated_plan)
+
+    @patch("ops.model.Container.exec")
+    @patch("ops.model.Container.exists")
+    def test_given_bessd_config_file_is_written_when_bessd_pebble_ready_then_initial_commands_are_executed(  # noqa: E501
+        self, patch_exists, patch_exec
+    ):
+        patch_exists.return_value = True
+
+        self.harness.container_pebble_ready(container_name="bessd")
+
+        patch_exec.assert_has_calls(
+            calls=[
+                call(
+                    command=["ip", "route", "replace", "192.168.251.0/24", "via", "192.168.252.1"],
+                    timeout=30,
+                    environment=None,
+                ),
+                call().wait_output(),
+                call(
+                    command=[
+                        "ip",
+                        "route",
+                        "replace",
+                        "default",
+                        "via",
+                        "192.168.250.1",
+                        "metric",
+                        "110",
+                    ],
+                    timeout=30,
+                    environment=None,
+                ),
+                call().wait_output(),
+                call(
+                    command=[
+                        "iptables",
+                        "-I",
+                        "OUTPUT",
+                        "-p",
+                        "icmp",
+                        "--icmp-type",
+                        "port-unreachable",
+                        "-j",
+                        "DROP",
+                    ],
+                    timeout=30,
+                    environment=None,
+                ),
+                call().wait_output(),
+            ],
+        )
+
+    @patch("ops.model.Container.exec")
+    @patch("ops.model.Container.exists")
+    def test_given_exec_error_when_bessd_pebble_ready_then_status_is_waiting(
```

Comment:
Is the config file missing from the test name?
```suggestion
    def test_given_config_file_is_exists_and_exec_error_when_bessd_pebble_ready_then_status_is_waiting(
```